cmd/vicadmin/vicadm.go

```diff
@@ -51,14 +51,19 @@ import (
 var (
 	logFileDir = "/var/log/vic/"
 
+	// constants for any kind of username/password based authentication
+	usernameEnvVar = "vicadmin_username"
+	passwordEnvVar = "vicadmin_password"
+
 	config struct {
 		session.Config
-		addr         string
-		dockerHost   string
-		vmPath       string
-		hostCertFile string
-		hostKeyFile  string
-		tls          bool
+		addr          string
+		dockerHost    string
+		vmPath        string
+		hostCertFile  string
+		hostKeyFile   string
+		authenticator authenticator
```

Comment:
I try to avoid name collisions of types and vars wherever possible.  Is it possible you want an embedded type?